### PR TITLE
Updating nimcache crd to auto detect

### PR DESCRIPTION
**Pull request by visheshtanksale**

### CODE REVIEW

**Review by slu2011 — APPROVED** [review]

**Review by shivamerla — DISMISSED** [review]
LGTM

File: api/apps/v1alpha1/nimcache_types.go [hunk 1]
```diff
@@ -71,8 +71,6 @@ type NGCSource struct {
 type ModelSpec struct {
 	// Profiles are the specific model profiles to cache
```
Comment (shivamerla) [inline api/apps/v1alpha1/nimcache_types.go:72]: We should make this comment more explicit along the lines of "Profiles indicate specific NIM profiles to cache. When these are provided, rest of the model parameters for profile selection are ignored"

**Comment by mikemckiernan** [comment]
Is the goal to select model profiles automatically, by default, unless the model profiles are specified?

**Comment by shivamerla** [comment]
> Is the goal to select model profiles automatically, by default, unless the model profiles are specified?

Yes, we discussed this, the default is to apply profile selection with given parameters, when these are selected explicitly. The `autoDetect` field was redundant in this case. This also means, by default the operator will detect GPUs and apply filtering when explicit GPUs are not provided for the optimized models.

File: api/apps/v1alpha1/nimcache_types.go (hunk 1 above)
Comment (visheshtanksale) [reply api/apps/v1alpha1/nimcache_types.go]: Fixed

**Review by shivamerla — DISMISSED** [review]
lgtm, thanks

**Review by shivamerla — APPROVED** [review]

**Pull request merged**